app/javascript/controllers/edit_deposit_controller.js

```diff
@@ -1,36 +1,29 @@
 import { Controller } from "stimulus";
 
 export default class extends Controller {
-  static targets = ["title", "titleField",
-                    "file", "fileField",
-                    "keywordsField",
-                    "contributorsField",
-                    "embargo-dateField",
-                    "terms", "termsField",
-                    "moreTypesLink", "moreTypes"];
+  static targets = ["moreTypesLink",
+                    "moreTypes",
+                    "frame"]
+  static values = { endpoint: String }
 
   connect() {
-    // TODO see what of the things are already valid
-    this.checkField(this.fileFieldTarget)
-    this.checkField(this.titleFieldTarget)
-    this.checkField(this.termsFieldTarget)
+    this.check()
+    this.element.addEventListener('change', () => this.check())
   }
 
   check(e) {
-    this.checkField(e.target)
-  }
-
-  checkField(field) {
-    const stepName = field.getAttribute("data-progress-step")
-    const step = this.targets.find(stepName)
-    let isComplete = field.value !== ''
-    if (stepName === 'file') {
-      isComplete = document.querySelectorAll('.dz-preview').length > 0
+    let form = this.element
+    let data = new FormData(form)
```

Comment:
Should these be `const`s? :man_shrugging:

Reply:
yep